### ISSUE TITLE
Document behavior for tslint.json in subfolders when config file is specified via CLI flag

### ISSUE THREAD

**Issue by Akuukis** [palantir/tslint#4762]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.3.4000
- __Running TSLint via__: CLI (and applies to VSCode too)

#### Repo structure

```yaml
packages/
    workspaces/
        package1/
            src/index.ts
            tslint.json
tslint.json
tsconfig.json
```

#### Actual behavior

```bash
yarn run tslint -p tsconfig.json                 # subfolder tslint.json takes precedence.
yarn run tslint -p tsconfig.json -c tslint.json  # subfolder tslint.json seems ignored.
```
#### Expected behavior

I expected that passing default value wont change the behavior.

#### Related issues

#4568
#4570

**Comment by adidahiya** [comment]
Can you elaborate on your use case? In your example it seems like the first command, `yarn run tslint -p tsconfig.json`, is sufficient.

**Comment by Akuukis** [comment]
@adidahiya Yes, that's enough for my use-case, but it was frustrating and took quite a time to figure out what caused the problem - I didn't expect that passing the default value would change the behavior, and I prefer to explicitly list parameters in my scripts. Not fixing but rather documenting the different would also solve my use-case.

I see another problem here - if someone is forced to use config file with non-default name, then currently it's impossible to achieve sub-folder behavior.

**Comment by adidahiya** [comment]
> I didn't expect that passing the default value would change the behavior

I don't think this expectation is so clear-cut; I could imagine someone arguing the opposite.

Also regarding a config file named something other than `tslint.json`... that seems like a rare / low priority use case. I'm open to other ideas if people feel strongly about this and have a concrete proposal on what to change, but for now I find the current behavior ok.

**Comment by Akuukis** [comment]
Ok, then let's document this difference?

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
